dev/src/lobster/flat_gltf.h

```diff
@@ -0,0 +1,789 @@
+            << std::endl;
+
+  // generate a VAO (1 per mesh)
+  glGenVertexArrays(1, &mesh.vao[0]);
+  // switch graphics memory to the VAO
+  glBindVertexArray(mesh.vao[0]);
+
+  _VERBOUT_ << "VAO address:           " << mesh.vao[0] << std::endl;
+
+  // turn on 6 attribute arrays
+  glEnableVertexAttribArray(0); // Position
+  glEnableVertexAttribArray(1); // Normal
+  glEnableVertexAttribArray(2); // UV
+  glEnableVertexAttribArray(3); // Color (not used)
+  glEnableVertexAttribArray(4); // Weights
+  glEnableVertexAttribArray(5); // Indices
+
+  // generate a vertex buffer and an index buffer
+  glGenBuffers(2, &mesh.vao[1]);
+
+  _VERBOUT_ << "Vertex buffer address: " << mesh.vao[1] << std::endl;
+  _VERBOUT_ << "Index buffer address:  " << mesh.vao[2] << std::endl;
+
+  // switch to the vertex buffer
+  glBindBuffer(GL_ARRAY_BUFFER, mesh.vao[1]);
+  // upload the vertices, passing the size in bytes
+  glBufferData(GL_ARRAY_BUFFER, vertices.size() * sizeof(Vertex), &vertices[0],
+               streaming ? GL_STREAM_DRAW : GL_STATIC_DRAW);
+  _VERBOUT_ << "Number of vertices: " << vertices.size() << std::endl;
+  _VERBOUT_ << "Size of Vertex:     " << sizeof(Vertex) << std::endl;
+
+  // perhaps an inconsistent use of data here but I'm not sure what to do
+  // sizeof returns a size_t which can be cast to anything since it's just a
+  // number
+  std::size_t pointer = 0;
+  glVertexAttribPointer(0, 4, GL_FLOAT, GL_FALSE, sizeof(Vertex),
+                        (const GLvoid *)pointer); // Position
+  pointer += sizeof(vertices[0].Position);
+  glVertexAttribPointer(1, 3, GL_FLOAT, GL_FALSE, sizeof(Vertex),
+                        (const GLvoid *)pointer); // Normal
+  pointer += sizeof(vertices[0].Normal);
+  glVertexAttribPointer(2, 2, GL_FLOAT, GL_FALSE, sizeof(Vertex),
+                        (const GLvoid *)pointer); // UV
+  pointer += sizeof(vertices[0].UV);
+  glVertexAttribPointer(3, 4, GL_FLOAT, GL_FALSE, sizeof(Vertex),
+                        (const GLvoid *)pointer); // Color (not used)
+  pointer += sizeof(vertices[0].Color);
+  glVertexAttribPointer(4, 4, GL_FLOAT, GL_FALSE, sizeof(Vertex),
+                        (const GLvoid *)pointer); // Weights
+  pointer += sizeof(vertices[0].Weights);
+  glVertexAttribPointer(5, 4, GL_FLOAT, GL_FALSE, sizeof(Vertex),
+                        (const GLvoid *)pointer); // Indices
+
+  // switch to the index buffer
+  glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, mesh.vao[2]);
+  // upload the indices, passing the size in bytes
+  glBufferData(GL_ELEMENT_ARRAY_BUFFER, indices.size() * sizeof(Index),
+               &indices[0], streaming ? GL_STREAM_DRAW : GL_STATIC_DRAW);
+  _VERBOUT_ << "Number of indices: " << indices.size() << std::endl;
+  _VERBOUT_ << "Size of Index:     " << sizeof(Index) << std::endl << std::endl;
+
+  // switch out of the VAO
+  glBindVertexArray(0);
+
+  // failure conditions are currently unimplemented
```

Comment:
The style of the rest of the codebase is not to use quite as many blank lines as you do. Also start comments with a captital and end with a `.`, i.e. a sentence.

Reply:
Comments fixed. As for the blank lines, I have a hard time scanning code so I'm not sure which blank lines should stay and which blank lines should go.